server_development/topics/providers.adoc

```diff
@@ -143,7 +141,7 @@ package org.acme.provider;
 
 import ...
 
-public class MyThemeSelectorProviderFactory implements ThemeSelectorProviderFactory, ServerInfoAwareProviderFactory {
+public class MyThemeSelectorProvider implements ThemeSelectorProvider, ServerInfoAwareProviderFactory {
```

Comment:
Thanks, Marek.   Maybe you can help me figure this out. This is not a change I made.
I see the change in my fork of the repository, but I don't make changes in the fork.  I only make changes in the clone of the fork.  

However, I see this change is already published here:
https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.3/html-single/server_developer_guide/index#providers_admin_console

Reply:
I've checked the github history a bit.

So we had this part broken in the keycloak documentation and also in the RHSSO 7.3 documentation as you pointed. Then one user from the community added the  fix, which fixes this incorrect example. It is from May this year: https://github.com/keycloak/keycloak-documentation/commit/061211234bfe7c64675c41762ee77a88ae983465 . In your fork, you basically reverted this older commit somehow.

IMO it may help you to always doublecheck if the changes in your PR are just changes you want. Also it is good practice to always do "git rebase" against latest upstream master before you send PR to make sure that you're not in conflict with latest master. I suggest to maybe take a look at some git tutorial for more details.

Reply:
Thanks, Marek for finding the problem.  I am still figuring out GIT although I was following a tutorial to do this and it included git rebase.  However, something went wrong.  I will fix it today.